### ISSUE TITLE
Alternative deployment methods?

### ISSUE THREAD

**Issue by maximcherny** [sensepost/snoopy-ng#39]
How awesome would it be to have snoopy deployed using something like this?

https://resin.io/how-it-works/

Thoughts?

**Comment by alexandrosm** [comment]
Hey there, resin.io founder here, noticed the link in our analytics. More than happy to help get snoopy working on resin.io and answer any questions you may have.

**Comment by maximcherny** [comment]
@alexandrosm g'day, resin.io does not seem to support 3G-based connectivity models yet, is that something that's on the roadmap?

**Comment by alexandrosm** [comment]
Hi Maxim, good day! 3G connectivity works today, I'd be interested in understanding how this impression was formed. If there is outdated material out there indicating this, I'd like to try and update it.

Resin works just fine over 3g, and we set up a VPN so we are able to reach devices behind the NAT that 3g providers usually have. We've also recently made improvements in how much bandwidth is needed to sustain a device in the network over a month. More improvements are coming here.

There will be significant improvement in download times when we release our container delta updates strategy soon. You can see here a video where we update 24 devices over a single 3g connection in about 2 minutes: https://www.youtube.com/watch?v=0n3J00ipZzw

In short, resin.io works with 3g just fine, and we're working on quite a few improvements that should make using resin even lighter on the data usage.

Happy to answer any more questions you may have.

**Comment by maximcherny** [comment]
@alexandrosm Thanks, I have only just scratched the surface... Basically, when I go and create a project the connectivity modes I am able to select are either ethernet or wi-fi (with the docs mentioning that wi-fi creds are subsequently hard to change at this stage) and 3g connectivity settings are not immediately obvious to me, hence the impression of its absence.

**Comment by alexandrosm** [comment]
Hi Maxim, sorry for dropping this, it actually started a pretty good discussion internally in the team. So first of all, you made a really good comment about our provisioning dialog giving a bad impression. We could indeed have 3G configuration options on that dialog and we've started the conversation on how that would work. On the other hand, people are already using resin.io with 3G dongles that just work. So in that case, just going with the Ethernet option and using a 3G dongle would be the way forward. And of course using a 3G router that interfaces via Ethernet is also possible. So while there are ways to use 3G (and people already do it) I do agree with you that we could make that more obvious, and also add the ability to configure 3G connectivity from the provisioning screen as well.
